main.go

```diff
@@ -0,0 +1,202 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"strconv"
+
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	"github.com/cloudogu/k8s-service-discovery/controllers"
+
+	// Import all Kubernetes client auth plugins (e.g. Azure, GCP, OIDC, etc.)
+	// to ensure that exec-entrypoint and run can make use of them.
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/healthz"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+	//+kubebuilder:scaffold:imports
+)
+
+const (
+	IngressClassName = "k8s-ecosystem-ces-service"
+)
+
+var (
+	scheme               = runtime.NewScheme()
+	setupLog             = ctrl.Log.WithName("setup")
+	metricsAddr          string
+	enableLeaderElection bool
+	probeAddr            string
+	// namespaceEnvVar is the constant for env variable WATCH_NAMESPACE
+	// which specifies the Namespace to the service discovery operator operates in.
+	// An empty value means the operator is running with cluster scope.
+	namespaceEnvVar = "WATCH_NAMESPACE"
```

Comment:
Maybe just call it NAMESPACE as there is no other namespace involved, e.g. there is no ACT_NAMESPACE or something like that.
Also mention that when running with cluster scope, the creation of ingress objects will fail when they are inside a namespace.

Reply:
There is still a difference between being in a namespace and watching one. The operator itself is not launched into the namespaced provided by namespaceEnvVar. The environment variable only contains the name of the namespace that should be observed by the service discovery. I rewrite the functionality by throwing an error if the watch namespace variable is missing.